sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClient.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Messaging.EventGrid.Models;
+
+namespace Azure.Messaging.EventGrid.Customization
+{
+    public class EventGridClient
+    {
+        private readonly ServiceRestClient _serviceRestClient;
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly string _hostName;
+        private const string SasKeyName = "aeg-sas-key";
+        //private const string SasTokenName = "aeg-sas-token";
+
+        /// <summary>Initalizes an instance of EventGridClient</summary>
+        protected EventGridClient()
+        {
+        }
+
+        /// <summary>Initalizes an instance of EventGridClient</summary>
+        /// <param name="endpoint">topic endpoint</param>
+        /// <param name="credential">used to connect to Azure</param>
+        public EventGridClient(Uri endpoint, AzureKeyCredential credential)
+            : this(endpoint, credential, new EventGridClientOptions())
+        {
+        }
+
+        /// <summary>Initalizes an instance of EventGridClient</summary>
+        /// <param name="endpoint">topic endpoint</param>
+        /// <param name="credential">used to connect to Azure</param>
+        /// <param name="options">configuring options</param>
+        public EventGridClient(Uri endpoint, AzureKeyCredential credential, EventGridClientOptions options)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+            _hostName = endpoint.Host;
+            HttpPipeline pipeline = HttpPipelineBuilder.Build(options, new AzureKeyCredentialPolicy(credential, SasKeyName));
+            _serviceRestClient = new ServiceRestClient(new ClientDiagnostics(options), pipeline);
+            _clientDiagnostics = new ClientDiagnostics(options);
+        }
+
+        /// <summary> Publishes a batch of EventGridEvents to an Azure Event Grid topic. </summary>
+        /// /// <param name="events"> An array of events to be published to Event Grid. </param>
```

Comment:
Nit - I don't think the doc comments will pick the param up if it's nested behind double `///`s.